cpp/tests/test_abm_location.cpp

```diff
@@ -509,30 +509,36 @@ TEST(TestLocation, storeCurrentSubpopulations)
     auto person3 = mio::abm::Person(location, mio::abm::InfectionState::Exposed, mio::abm::AgeGroup::Age35to59, {});
     location.add_person(person3);
 
-    auto t1 = mio::abm::TimePoint(0) + mio::abm::days(7);
+    auto t1 = mio::abm::TimePoint(0) + mio::abm::hours(7);
     location.add_subpopulations_timepoint(t1);
-    auto v1 = location.get_population().get_value(0);
-    ASSERT_EQ(v1[mio::abm::InfectionState::Infected], 2); // Check whether the number of persons in infected state at the location is correct
+    auto v1 = location.get_population().get_value(1);
+    // Check whether the number of persons in infected state at the location is correct
+    ASSERT_EQ(v1[int(mio::abm::InfectionState::Infected)], 2);
 
-    auto t2 = mio::abm::TimePoint(0) + mio::abm::days(14);
+    auto t2 = mio::abm::TimePoint(0) + mio::abm::hours(14);
     person1.set_infection_state(mio::abm::InfectionState::Infected_Critical);
     location.changed_state(person1, mio::abm::InfectionState::Infected);
     location.add_subpopulations_timepoint(t2);
-    auto v2 = location.get_population().get_value(1);
-    ASSERT_EQ(v2[mio::abm::InfectionState::Infected], 1); // Check whether the number of persons in infected state at the location is correct
+    auto v2 = location.get_population().get_value(2);
+    // Check whether the number of persons in infected state at the location is correct
+    ASSERT_EQ(v2[int(mio::abm::InfectionState::Infected)], 1);
 
-    auto t3 = mio::abm::TimePoint(0) + mio::abm::days(24);
+    auto t3 = mio::abm::TimePoint(0) + mio::abm::hours(24);
     person3.set_infection_state(mio::abm::InfectionState::Infected);
     location.changed_state(person3, mio::abm::InfectionState::Exposed);
     location.add_subpopulations_timepoint(t3);
-    auto v3 = location.get_population().get_value(2);
-    ASSERT_EQ(v3[mio::abm::InfectionState::Infected], 2); // Check whether the number of persons in infected state at the location is correct
+    auto v3 = location.get_population().get_value(3);
+    // Check whether the number of persons in infected state at the location is correct
+    ASSERT_EQ(v3[int(mio::abm::InfectionState::Infected)], 2);
```

Comment:
```suggestion
    ASSERT_EQ(v3[size_t(mio::abm::InfectionState::Infected)], 2);
```
I think it is an unsigned index, not sure though

Reply:
Yes, it is fixed!